packages/autorest.python/autorest/codegen/serializers/builder_serializer.py

```diff
@@ -929,10 +931,19 @@ def response_headers_and_deserialization(
                 )
             )
         elif response.type:
-            if self.code_model.options["models_mode"]:
+            if self.code_model.options["models_mode"] == "msrest":
                 retval.append(
                     f"deserialized = self._deserialize('{response.serialization_type}', pipeline_response)"
                 )
+            elif self.code_model.options["models_mode"] == "dpg":
+                if isinstance(response.type, ModelType):
+                    retval.append(
+                        f"deserialized = _models.{response.type.name}(pipeline_response.http_response.json())"
```

Comment:
nit: you can make a var for `pipeline_response.http_response.json()` so you don't have to type it out twice